### PR TITLE
hw5.1

### PR DESCRIPTION
**Pull request by Mykhaylo12**

### CODE REVIEW

File: src/main/java/core/basesyntax/Car.java [hunk 1]
```diff
@@ -1,5 +1,71 @@
...
+        this.modelCar = modelCar;
+        this.maxSpedCar = maxSpedCar;
+        this.ageCar = ageCar;
+        this.weightCar = weightCar;
+        this.engine = engine;
+    }
+
+    @Override
+    protected Object clone() throws CloneNotSupportedException {
+        return super.clone();
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Car.java]: try to solve without super.clone()

File: src/main/java/core/basesyntax/Car.java [hunk 2]
```diff
@@ -1,5 +1,71 @@
...
+    }
+
+    public Car(Car clone) {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Car.java]: remove this constructor, please

File: src/main/java/core/basesyntax/Engine.java [hunk 3]
```diff
@@ -1,5 +1,60 @@
...
+                  int ageEngine, int weightEngine) {
+        this.nameEngine = nameEngine;
+        this.modelEngine = modelEngine;
+        this.volumeEngine = volumeEngine;
+        this.ageEngine = ageEngine;
+        this.weightEngine = weightEngine;
+    }
+
+    @Override
+    protected Object clone() throws CloneNotSupportedException {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Engine.java]: same

File: src/main/java/core/basesyntax/Car.java (hunk 2 above)
Comment (Mykhaylo12) [reply src/main/java/core/basesyntax/Car.java]: ok

File: src/main/java/core/basesyntax/Car.java [hunk 4]
```diff
@@ -1,5 +1,65 @@
...
+               int weightCar, Engine engine) {
+        this.nameCar = nameCar;
+        this.modelCar = modelCar;
+        this.maxSpedCar = maxSpedCar;
+        this.ageCar = ageCar;
+        this.weightCar = weightCar;
+        this.engine = engine;
+    }
+
+    public Car clone() {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Car.java]: we should override this method, not create new one

File: src/main/java/core/basesyntax/Car.java (hunk 1 above)
Comment (Mykhaylo12) [reply src/main/java/core/basesyntax/Car.java]: done

File: src/main/java/core/basesyntax/Car.java (hunk 4 above)
Comment (Mykhaylo12) [reply src/main/java/core/basesyntax/Car.java]: done

Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Car.java]: where is `@Override`?

File: src/main/java/core/basesyntax/Car.java [hunk 5]
```diff
@@ -1,5 +1,65 @@
...
+        return new Car(nameCar, modelCar, maxSpedCar, ageCar, weightCar, engine);
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Car.java]:
```suggestion
        return new Car(nameCar, modelCar, maxSpedCar, ageCar, weightCar, engine.clone());
```

**Pull request closed without merging**